### ISSUE TITLE
Incorrect delegation from PrintObj to ViewObj, and from PrintString to ViewString

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/Semigroups#518]
Semigroups currently violates the delegation rules for printing and viewing described in [section 6.3 of the GAP reference manual](https://www.gap-system.org/Manuals/doc/ref/chap6.html#X8074A8387C9DB9A8). Overall, it shouldn't be difficult to fix this, I think. Here is the report generated by `CheckOutputDelegations()` (in the fixed version from <https://github.com/gap-system/gap/pull/2816>). For the first case, you just need to switch the `ViewObj` and `PrintObj` declarations. For the second, replacing `ViewString` by `PrintString` in the function should do it.
```
gap> Read("tst/testutil.g");
gap> CheckOutputDelegations();
All delegations correct for Display
All delegations correct for ViewObj

Detected incorrect delegations for PrintObj
---------------------------------------------------------------
PrintObj: for a free inverse semigroup element
 delegates to [ "ViewObj" ]
Filename: /Users/mhorn/Projekte/GAP/gap.github/pkg/semigroups-3.0.16/gap/fp/freeinverse.gi:206
<Operation "ViewObj">
---------------------------------------------------------------
All delegations correct for DisplayString
All delegations correct for ViewString

Detected incorrect delegations for PrintString
---------------------------------------------------------------
PrintString: for a pbr
 delegates to [ "ViewString" ]
Filename: /Users/mhorn/Projekte/GAP/gap.github/pkg/semigroups-3.0.16/gap/elements/pbr.gi:698
function ( x )
    local ext, str;
    ext := ExtRepOfObj( x );
    str := Concatenation( "\>\>PBR(\>\>", ViewString( ext[1] ), "\<\<," );
    if Length( String( ext[1] ) ) > 72 or Length( String( ext[2] ) ) > 72 then
        Append( str, "\n" );
    else
        Append( str, " " );
    fi;
    Append( str, "\>\>" );
    Append( str, ViewString( ext[2] ) );
    Append( str, "\<\<\<\<)" );
    return str;
end
---------------------------------------------------------------
All delegations correct for String
```

**Comment by wilfwilson** [comment]
Thanks @fingolfin, it's time for me to learn how this stuff is supposed to work! I've implemented your suggestions and running `CheckOutputDelegations()` from your branch 
https://github.com/gap-system/gap/pull/2816 works cleanly now.

**Comment by wilfwilson** [comment]
Resolved by #521.
